src/Centreon/Domain/HostConfiguration/Model/HostGroup.php

```diff
@@ -0,0 +1,373 @@
+     */
+    private $notes;
+
+    /**
+     * @var string|null Define an optional URL that can be used to provide more information about the host group.
+     * <br> Any valid URL can be used.
+     * <br> This can be very useful if you want to make detailed information on the host group,
+     * emergency contact methods, etc. available to other support staff.
+     */
+    private $notesUrl;
+
+    /**
+     * @var string|null Define an optional URL that can be used to provide more actions to be performed on
+     * the host group. You will see the link to the action URL in the host group details.
+     */
+    private $actionUrl;
+
+    /**
+     * @var Image|null Define the image that should be associated with this host group.
+     * This image will be displayed in the various places. The image will look best if it is 40x40 pixels in size.
+     */
+    private $icon;
+
+    /**
+     * @var Image|null Define an image that should be associated with this host group in the statusmap CGI
+     * in monitoring engine. <br>
+     * You can choose a JPEG, PNG, and GIF image. The GD2 image format is preferred, as other image formats
+     * must be converted first when the statusmap image is generated. <br>
+     * The image will look best if it is 40x40 pixels in size.
+     */
+    private $iconMap;
+
+    /**
+     * @var string|null Geographical coordinates use by Centreon Map module to position element on map. <br>
+     * Define "Latitude,Longitude", for example for Paris coordinates set "48.51,2.20"
+     */
+    private $geoCoords;
+
+    /**
+     * @var int|null RRD retention duration (in days) of all the services that are in this host group.
+     * If service is in multiple host groups, the highest retention value will be used.
+     */
+    private $rrd;
+
+    /**
+     * @var string|null Comments on this host group.
+     */
+    private $comment;
+
+    /**
+     * @var bool Indicates whether the host group is activated or not.
+     */
+    private $isActivated = true;
+
+    /**
+     * @param string $name Host Group name
+     * @throws \Assert\AssertionFailedException
+     */
+    public function __construct(string $name)
+    {
+        $this->setName($name);
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     * @return HostGroup
+     */
+    public function setId(int $id): HostGroup
+    {
+        $this->id = $id;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     * @return HostGroup
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setName(string $name): HostGroup
+    {
+        if ($name !== null) {
+            Assertion::maxLength($name, self::MAX_NAME_LENGTH, 'HostGroup::name');
+        }
```

Comment:
in method definition $name can't be null , I don't really know if you have to remove this condition, or add string|null to property name (and adequate changes in getter/setter), depending of the needed behaviour.